src/datetime.js

```diff
@@ -745,8 +745,10 @@ export default class DateTime {
    * @param {string} [opts.locale='system\'s locale'] - a locale to set on the resulting DateTime instance
    * @param {string} opts.outputCalendar - the output calendar to set on the resulting DateTime instance
    * @param {string} opts.numberingSystem - the numbering system to set on the resulting DateTime instance
+   * @param {DateTime|Date|Object} opts.referenceDate - the reference date to take for missing parts
    * @example DateTime.fromObject({ year: 1982, month: 5, day: 25}).toISODate() //=> '1982-05-25'
    * @example DateTime.fromObject({ year: 1982 }).toISODate() //=> '1982-01-01'
+   * @example DateTime.fromObject({ year: 1982 }, { referenceDate: { day: 10 } }).toISODate() //=> '1982-01-10'
```

Comment:
This example is actually wrong. The referenceDate is only used for units above the highest unit in the input. If `{ month: 6 }` is given, _only_ the year is taken from the reference date.

Reply:
Oh sorry, I just expected the _old_ `tsNow` to add all the missing parts (not just the units above). What's your opinion here? Change the logic on a passed refDate, so **all** missing parts are replaced? And if, always (could be a breaking change?) or just with passed refDate?

date-fns even uses proximity to guess the right input: https://date-fns.org/v3.6.0/docs/parse

Reply:
If that is added, it definitely needs to be opt in. `DateTime.fromObject({ year: 2013 })` shouldn't suddenly have the current time.
What's your use-case for this?

Reply:
Tbh, I have no use-case for it. Just thinking, in an Englisch format, if only this first digits (month) is set, maybe someone wants to have a different day than the first (and not only a different year than todays?). But again, no hard feeling here! Can just stay as it is!

Reply:
Let's keep it as it is for now then!

Reply:
The example here still contradicts the actual behavior and implementation.